src/exception/exception-repo.ts

```diff
@@ -35,58 +34,61 @@ const programExceptionSchema = new mongoose.Schema({
   ],
 });
 
-type ProgramExceptionDocument = mongoose.Document & ProgramException;
-
-export const ProgramExceptionModel = mongoose.model<ProgramExceptionDocument>(
+export const ProgramExceptionModel = mongoose.model<ProgramException>(
   'ProgramException',
   programExceptionSchema,
 );
 
+type RepoResponse = Promise<ProgramException | undefined>;
+
 export interface ProgramExceptionRepository {
-  save(exception: ProgramException): Promise<DeepReadonly<ProgramException>>;
-  find(programId: string): Promise<DeepReadonly<ProgramException> | undefined>;
-  delete(programId: string): Promise<void>;
+  save(exception: ProgramException): RepoResponse;
+  find(programId: string): RepoResponse;
+  delete(programId: string): RepoResponse;
 }
 
+const checkDoc = (doc: null | ProgramException): undefined | ProgramException => {
+  if (doc === null || (isArray(doc) && doc.length === 0)) {
+    return undefined;
```

Comment:
no need to change it, as this works perfectly, it caught me by surprise like with the "automatic semicolon insertion" battles, I didn't realise I'm more inclined towards `return;` than explicitly returning an `undefined`.
🤷 
Now I know.